job-server/src/main/scala/spark/jobserver/HealthCheck.scala

```diff
@@ -0,0 +1,81 @@
+package spark.jobserver
+
+import akka.actor.ActorRef
+import akka.pattern.ask
+import akka.util.Timeout
+import org.slf4j.LoggerFactory
+import scala.concurrent.duration._
+import scala.concurrent.{Await, Future}
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.util.control.NonFatal
+import spray.httpx.SprayJsonSupport.sprayJsonMarshaller
+import spray.json.DefaultJsonProtocol._
+import spray.http.StatusCodes
+import spray.routing.{HttpService, RequestContext}
+import spark.jobserver.io.JobInfo
+import spark.jobserver.io.JobDAOActor
+
+trait HealthCheckService {
+
+  def healthCheckImpl(ctx: RequestContext, supervisor: ActorRef, jobInfo: ActorRef,
```

Comment:
`healthCheckImpl` is a strange name for a function. It doesn't tell me what it is trying to do.

`isHealthy` seems to be more appropriate with a return type of `boolean`.
I do not like that we have to pass the request object down to this class. This function should just return a boolean and the `WebApi` should convert the boolean to `ctx.complete(....)`

Reply:
Further, the arguments that you are taking in are very specific to your implementation of actors health check. Can we improve this?

Reply:
I am fine with renaming the function and changing the return type. But I do not understand your comment about arguments. Can you explain little more about arguments?

Reply:
I meant that the implementations of HealthCheckService can vary. Some might have an implementation to check jobserver + hdfs + zookeeper others might just have a check for actors. The arguments seem specific to your implementation. I was saying that the trait function arguments should be a bit more generic.

Reply:
Addressed